bin/ffmpeg.py

```diff
@@ -87,7 +81,7 @@ def get_video_info(self, input_video):
         # this execution command needs to be hard-coded
         # since video2x only strictly recignizes this one format
         execute = [
-            self.ffmpeg_probe_binary,
+            self.settings['path'] / self.settings['ffprobe_binary'],
```

Comment:
I don't quite follow what you mean here.

self.settings['path'] is dynamic based on what find_path finds.  self.settings['ffprobe_binary'] is too, eg with a .exe

Technically it could be self.settings['ffprobe_path'] and self.settings['ffmpeg_path'] but that wouldn't line up with the way the rest of the code base.  Uniform style grants the ability to easily detect bugs by looking at syntax, as well as ease both refactoring and implementing feature requests.  Because of this I highly recommend keeping a standard style as a principle to aim for, but not necessarily a principle to strictly follow.

Reply:
What I mean here is that you're using `self.settings['path'] / self.settings['ffprobe_binary']` multiple times in the this file.

This means each time a ffmpeg command is executed, the full path object will be reevaluated. It makes more sense to me to generate a static path object for the full path of the binary file for each Ffmpeg class instance.

Reply:
Therfore, `self.ffmpeg_binary = self.settings['path'] / self.settings['ffmpeg_binary']` during `__init__`

Reply:
It's doable, but I worry it could fall into premature optimization.

Reply:
honestly, it's just the simple fact that this evaluation occurs so many times in the file seems wrong to me

Reply:
I feel yah.  I'll change it.

If you're curious, my general rule of thumb is I follow a loose rule of three.  Basically, is there something in the code base that appears 3 or more times?  (Similar classes, similar functions, or similar lines of code (eg `execute = [str(e) for e in execute]`), and so on.)  Then I consider refactoring.  Because it is only happening twice I disregarded the notion of refactoring those lines.